zipline/assets/assets.py

```diff
@@ -138,6 +184,72 @@ def _reset_caches(self):
         for cache in self._caches:
             cache.clear()
 
+    def reload_symbol_maps(self):
+        """Clear the in memory symbol lookup maps.
+
+        This will make any changes to the underlying db available to the
+        symbol maps.
+        """
+        # clear the lazyval caches, the next access will requery
+        del type(self).symbol_ownership_map[self]
+        del type(self).fuzzy_symbol_ownership_map[self]
+
+    @lazyval
+    def symbol_ownership_map(self):
+        rows = sa.select(self.equity_symbol_mappings.c).execute().fetchall()
+
+        mappings = {}
+        for row in rows:
+            mappings.setdefault(
+                (row.company_symbol, row.share_class_symbol),
+                [],
+            ).append(
+                SymbolOwnership(
+                    pd.Timestamp(row.start_date, unit='ns', tz='utc'),
+                    pd.Timestamp(row.end_date, unit='ns', tz='utc'),
+                    row.sid,
+                    row.symbol,
+                ),
+            )
+
+        return valmap(
+            lambda v: tuple(
+                SymbolOwnership(
+                    a.start,
+                    b.start,
+                    a.sid,
+                    a.symbol,
+                ) for a, b in sliding_window(
+                    2,
+                    concatv(
+                        sorted(v),
+                        # concat with a fake ownership object to make the last
+                        # end date be max timestamp
```

Comment:
Right, I guess what I mean to ask is why do we want the max time instead of the end_date of the last equity to hold the ticker?